### PR TITLE
TYR: handle autocomplete auto update

### PR DESCRIPTION
**Pull request by antoine-de**
watch a directory and run either bano2mimir or osm2mimir when data appears.

linked to http://jira.canaltp.fr/browse/NAVP-548

the instance parameters are not yet handled

---

This change is [Reviewable](https://reviewable.io/reviews/canaltp/navitia/1541)

### CODE REVIEW

**Comment by buildbotctp** [comment]
Build finished.

File: source/tyr/tyr/tasks.py [hunk 1]
```diff
@@ -144,6 +147,103 @@ def update_data():
...
+    filename can be either a directory, a file or a list of files
+
+    return can be:
+        - 'bano'
+        - 'osm'
+
+    """
+    def files_type(files):
+        #first we try fusio, because it can load fares too
+        if any(f for f in files if BANO_REGEXP.match(f)):
```
Comment (kinnou02) [inline source/tyr/tyr/tasks.py:166]: We should be able to handle bano file like this: "full.csv.gz"

File: source/tyr/tyr/tasks.py [hunk 2]
```diff
@@ -144,6 +147,103 @@ def update_data():
...
+        if dataset.type in task:
+            if backup_file:
+                filename = move_to_backupdirectory(_file, autocomplete_instance.backup_dir(autocomplete_dir))
+            else:
+                filename = _file
+            actions.append(task[dataset.type].si(autocomplete_instance,
+                                                 filename=filename, dataset_uid=dataset.uid))
+        else:
+            #unknown type, we skip it
+            current_app.logger.debug("unknwn file type: {} for file {}"
```
Comment (kinnou02) [inline source/tyr/tyr/tasks.py]: typo

File: source/tyr/tyr/tasks.py (hunk 1 above)
Comment (antoine-de) [inline source/tyr/tyr/tasks.py:166]: done by changing the BANO regexp to _bano_, the .gz will be handled by bano2mimir

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**